### PR TITLE
monitor TKGS cluster class Cluster creation progress

### LINKED ISSUE #2604: TKGS Cluster Class based workload cluster creation progress monitor
(This is used to request new product features)

**Describe the feature request**
We need to implement the monitoring logic for TKGS Cluster Class based cluster creation, similar to how we monitor CAPI based resources for other infrastructures.
**Describe alternatives you've considered**
Disable the cluster creation monitoring logic for the TKGS Cluster Class-based cluster creation
**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [X] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by chandrareddyp**
### What this PR does / why we need it
This PR change monitors the TKGS clusterclass-based cluster creation progress, it uses existing CAPI cluster creation monitoring logic to monitor the TKGS classy cluster objects like Cluster, KubeadmControlPlaneList, MachineDeploymentList, MachineList, packages etc.

### Which issue(s) this PR fixes

Fixes #2604

### Describe testing done for PR

- TKGS Classy Cluster creation:
`tanzu cluster create -f tkgs-classy-cluster.yaml -v 9`
output:
```
Validating configuration...
Waiting for the Tanzu Kubernetes Cluster service for vSphere workload cluster
Waiting for cluster to be initialized...
Getting secret for cluster
Waiting for resource tkc-05-kubeconfig of type *v1.Secret to be up and running
Waiting for cluster nodes to be available...
Waiting for resource tkc-05 of type *v1beta1.Cluster to be up and running
Waiting for resources type *v1beta1.KubeadmControlPlaneList to be up and running
Waiting for resources type *v1beta1.MachineDeploymentList to be up and running
Waiting for resources type *v1beta1.MachineList to be up and running

Workload cluster 'tkc-05' created
```

- TKGS TKC Cluster creation:
`tanzu cluster create -f tkgs-tkc.conf`
output:
```
Validating configuration...
Waiting for the Tanzu Kubernetes Cluster service for vSphere workload cluster

Workload cluster 'tkc-05' created
```

### Release note

```release-note

```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by anujc25 — APPROVED** [review]
LGTM

File: pkg/v1/tkg/client/cluster.go [hunk 1]
```diff
@@ -468,6 +470,10 @@ func (c *TkgClient) createPacificCluster(options *CreateClusterOptions, waitForC
 		return nil
 	}
 
+	if options.IsInputFileClusterClassBased {
+		return c.waitForClusterCreation(clusterClient, options, options.IsInputFileClusterClassBased)
+	}
+
 	log.V(3).Infof("Waiting for the Tanzu Kubernetes Cluster service for vSphere workload cluster\n")
 
 	if err := clusterClient.WaitForPacificCluster(clusterName, namespace); err != nil {
```
Comment (vuil) [inline pkg/v1/tkg/client/cluster.go:479]: Is the if case supposed to replace this case? iow should L479 be in the else clause?

**Comment by vuil** [comment]
Please use "commit -s " to sign off the commit you posted, and eliminate the (fixes #...) from the header. It does not belong there. 
Feel free to include the description of the commit message.
Please include a brief description of what the change is supposed to do.

File: pkg/v1/tkg/client/cluster.go (hunk 1 above)
Comment (chandrareddyp) [reply pkg/v1/tkg/client/cluster.go:479]: Yes, that makes sense keep existing logic in else case and validate the error and return, I will update the code.

**Review by vuil — APPROVED** [review]
lgtm. thanks
please include testing result for the TKC case as well.

File: pkg/v1/tkg/client/cluster.go [hunk 2]
```diff
@@ -259,15 +259,17 @@ func (c *TkgClient) waitForClusterCreation(regionalClusterClient clusterclient.C
...
+			workloadClusterClient: workloadClusterClient,
+			clusterName:           options.ClusterName,
+			namespace:             options.TargetNamespace,
+			waitForCNI:            true,
+		}); err != nil {
+			return errors.Wrap(err, "error waiting for addons to get installed")
+		}
 	}
 
 	log.Info("Waiting for packages to be up and running...")
```
Comment (prkalle) [inline pkg/v1/tkg/client/cluster.go:275]: Question: Is this applicable for TKGs cluster?

**Review by prkalle — COMMENTED** [review]
LGTM.  Thanks
I have a question regarding waiting for packages.

File: pkg/v1/tkg/client/cluster.go (hunk 2 above)
Comment (chandrareddyp) [reply pkg/v1/tkg/client/cluster.go]: @prkalle We don't need to monitor packages for TKGS Cluster for now, I did changes to disable it.

**Pull request merged**